### PR TITLE
boards/nucleo144-f413 add inital support

### PR DESCRIPTION
**Pull request by vincent-d**
This adds support for nucleo144-f413 (mainly imported from #6553) and support for stm32f413.

It needs a patched openOCD because the f413 is not correctly detected. I'll try to submit my patch upstream ASAP.

### CODE REVIEW

**Comment by vincent-d** [comment]
OpenOCD patch is here http://openocd.zylin.com/#/c/3969/

File: boards/nucleo144-f413/board.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2016 Inria
```
Comment (aabadie) [inline boards/nucleo144-f413/board.c]: 2017, OTAKeys with Inria ?

File: boards/nucleo144-f413/board.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+/**
+ * @ingroup     boards_nucleo144-f413
+ * @{
+ *
+ * @file
+ * @brief       Board specific implementations for the nucleo144-f413 board
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
```
Comment (aabadie) [inline boards/nucleo144-f413/board.c]: You can add yourself in the list .

File: boards/nucleo144-f413/include/board.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2016 Inria
```
Comment (aabadie) [inline boards/nucleo144-f413/include/board.h]: Same comment and probably in other places.

File: boards/nucleo144-f413/include/periph_conf.h [hunk 4]
```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (C) 2016 Inria
+ * Copyright (C) 2017 OTA keys S.A.
```
Comment (aabadie) [inline boards/nucleo144-f413/include/periph_conf.h]: Ok :)

File: cpu/stm32f4/vectors.c [hunk 5]
```diff
@@ -108,6 +108,10 @@ WEAK_DEFAULT void isr_otg_hs_ep1_in(void);
 WEAK_DEFAULT void isr_otg_hs_wkup(void);
 WEAK_DEFAULT void isr_otg_hs(void);
 WEAK_DEFAULT void isr_dcmi(void);
+WEAK_DEFAULT void isr_can3_tx(void);
```
Comment (aabadie) [inline cpu/stm32f4/vectors.c]: maybe protect with the same if defined as below or the build will fail for other boards (not sure).

File: boards/nucleo144-f413/board.c (hunk 1 above)
Comment (vincent-d) [inline boards/nucleo144-f413/board.c]: As I didn't change the code from you PR, I didn't think about the copyright. Addressed

File: boards/nucleo144-f413/board.c (hunk 2 above)
Comment (vincent-d) [inline boards/nucleo144-f413/board.c]: Addressed

File: boards/nucleo144-f413/include/board.h (hunk 3 above)
Comment (vincent-d) [inline boards/nucleo144-f413/include/board.h]: Addessed

File: cpu/stm32f4/vectors.c (hunk 5 above)
Comment (vincent-d) [inline cpu/stm32f4/vectors.c]: Addressed, and add a comment with the reference manual

File: cpu/stm32f4/vectors.c [hunk 6]
```diff
@@ -103,11 +105,20 @@ WEAK_DEFAULT void isr_dma2_stream7(void);
 WEAK_DEFAULT void isr_usart6(void);
 WEAK_DEFAULT void isr_i2c3_ev(void);
 WEAK_DEFAULT void isr_i2c3_er(void);
+#if defined(CPU_MODEL_STM32F413ZH)
```
Comment (aabadie) [inline cpu/stm32f4/vectors.c]: Finally, I think the specific defined is not required. I tested it for another CPU (F0) and it still builds if the function is not used.
I don't know what's the best practice here. Maybe @haukepetersen can help ?

**Comment by vincent-d** [comment]
Cleaned up vendor header (remove trailing spaces)

**Comment by aabadie** [comment]
Looks good now. Can you squash the commits ?

**Comment by vincent-d** [comment]
Squashed

**Comment by vincent-d** [comment]
rebased

**Comment by aabadie** [comment]
Even if it's all green, I'm not sure if I should press the green button now : I prefer waiting for your [OpenOCD patch](http://openocd.zylin.com/#/c/3969/) to be merged first

**Comment by vincent-d** [comment]
We will need the cpu support soon for our hardware, so I would like to have this merged if possible.

**Comment by aabadie** [comment]
I see. Can you verify that the documentation in the wiki is ok ? In particular, the ["flashing" section](https://github.com/RIOT-OS/RIOT/wiki/Board%3A-Nucleo144-F413#flashing-the-device).
If you are ok, I'll merge.

**Comment by vincent-d** [comment]
wiki looks OK

**Comment by aabadie** [comment]
> wiki looks OK

then go !

**Pull request merged**